File: src/main/java/com/example/model/ExamCenters.java
package com.example.model;

import com.example.database.Database;
import com.example.database.ParamLength;
import java.sql.ResultSet;
import java.util.ArrayList;
import java.util.List;

public class ExamCenters {
    // Add these constants
    private static final int INPUT = 1;
    private static final int OUTPUT = 2;
    private static final int INPUT_OUTPUT = 3;
    
    // Then update all ParameterDirection references to use these constants
    // For example:
    public ResultSet findNearestExamCenter(String connectionString, int pincode, int examBodyId) throws Exception {
        Database objDatabase = null;
        ResultSet objResultSet = null;
        int procReturnValue = 0;
        List<Object> allParameters = new ArrayList<>();
        
        try {
            String procedureName = "STP_CMN_FindNearestExamCenter2";
            String[] params = new String[] { "@intPINCode", "@tntExamBodyID" };
            int[] paramTypes = new int[] { java.sql.Types.INTEGER, java.sql.Types.TINYINT };
            ParamLength[] paramLengths = new ParamLength[] { 
                new ParamLength(4, 10, 0), 
                new ParamLength(1, 3, 0) 
            };
            // Change this line:
            int[] paramDirections = new int[] { INPUT, INPUT };
            // ... rest of your code ...
            Object[] values = new Object[] { pincode, examBodyId };
            
            objDatabase = new Database();
            ResultSet[] resultSetWrapper = new ResultSet[1];
            
            procReturnValue = objDatabase.execProcedure(
                connectionString, procedureName, params, paramTypes, paramLengths, 
                paramDirections, values, allParameters, resultSetWrapper, true);
            
            objResultSet = resultSetWrapper[0];
        } catch (Exception ex) {
            throw ex;
        } finally {
            objDatabase = null;
        }
        
        return objResultSet;
    }
    
    public ResultSet getExamCenters(String connectionString, int pincode, int examBodyId) throws Exception {
        Database objDatabase = null;
        ResultSet objResultSet = null;
        int procReturnValue = 0;
        List<Object> allParameters = new ArrayList<>();
        
        try {
            String procedureName = "STP_CMN_FindNearestExamCenter3";
            String[] params = new String[] { "@intPINCode", "@tntExamBodyID" };
            int[] paramTypes = new int[] { java.sql.Types.INTEGER, java.sql.Types.TINYINT };
            ParamLength[] paramLengths = new ParamLength[] { 
                new ParamLength(4, 10, 0), 
                new ParamLength(1, 3, 0) 
            };
            int[] paramDirections = new int[] { ParameterDirection.INPUT, ParameterDirection.INPUT };
            Object[] values = new Object[] { pincode, examBodyId };
            
            objDatabase = new Database();
            ResultSet[] resultSetWrapper = new ResultSet[1];
            
            procReturnValue = objDatabase.execProcedure(
                connectionString, procedureName, params, paramTypes, paramLengths, 
                paramDirections, values, allParameters, resultSetWrapper, true);
            
            objResultSet = resultSetWrapper[0];
        } catch (Exception ex) {
            throw ex;
        } finally {
            objDatabase = null;
        }
        
        return objResultSet;
    }
    
    public ResultSet examCentersForState(String connectionString, int stateId, boolean isTbxCenter) throws Exception {
        Database objDatabase = null;
        ResultSet objResultSet = null;
        int procReturnValue = 0;
        List<Object> allParameters = new ArrayList<>();
        
        try {
            String procedureName = "sp_get_examcenters";
            String[] params = new String[] { "@stateid", "@IsTBXCenter" };
            int[] paramTypes = new int[] { java.sql.Types.INTEGER, java.sql.Types.VARCHAR };
            ParamLength[] paramLengths = new ParamLength[] { 
                new ParamLength(4, 10, 0), 
                new ParamLength(1, 0, 0) 
            };
            int[] paramDirections = new int[] { ParameterDirection.INPUT, ParameterDirection.INPUT };
            Object[] values = new Object[] { stateId, isTbxCenter ? "Y" : "N" };
            
            objDatabase = new Database();
            ResultSet[] resultSetWrapper = new ResultSet[1];
            
            procReturnValue = objDatabase.execProcedure(
                connectionString, procedureName, params, paramTypes, paramLengths, 
                paramDirections, values, allParameters, resultSetWrapper, true);
            
            objResultSet = resultSetWrapper[0];
        } catch (Exception ex) {
            throw ex;
        } finally {
            objDatabase = null;
        }
        
        return objResultSet;
    }
    
    public ResultSet examCentersForState(String connectionString, int stateId, int centerId) throws Exception {
        Database objDatabase = null;
        ResultSet objResultSet = null;
        int procReturnValue = 0;
        List<Object> allParameters = new ArrayList<>();
        
        try {
            String procedureName = "sp_get_examcenters2";
            String[] params = new String[] { "@stateid", "@centerid" };
            int[] paramTypes = new int[] { java.sql.Types.INTEGER, java.sql.Types.INTEGER };
            ParamLength[] paramLengths = new ParamLength[] { 
                new ParamLength(4, 10, 0), 
                new ParamLength(4, 10, 0) 
            };
            int[] paramDirections = new int[] { ParameterDirection.INPUT, ParameterDirection.INPUT };
            Object[] values = new Object[] { 
                stateId <= 0 ? null : stateId, 
                centerId <= 0 ? null : centerId 
            };
            
            objDatabase = new Database();
            ResultSet[] resultSetWrapper = new ResultSet[1];
            
            procReturnValue = objDatabase.execProcedure(
                connectionString, procedureName, params, paramTypes, paramLengths, 
                paramDirections, values, allParameters, resultSetWrapper, true);
            
            objResultSet = resultSetWrapper[0];
        } catch (Exception ex) {
            throw ex;
        } finally {
            objDatabase = null;
        }
        
        return objResultSet;
    }
    
    public ResultSet examCentersForStateEx(String connectionString, int stateId, int centerId) throws Exception {
        Database objDatabase = null;
        ResultSet objResultSet = null;
        int procReturnValue = 0;
        List<Object> allParameters = new ArrayList<>();
        
        try {
            String procedureName = "sp_get_examcenters5";
            String[] params = new String[] { "@stateid", "@centerid" };
            int[] paramTypes = new int[] { java.sql.Types.INTEGER, java.sql.Types.INTEGER };
            ParamLength[] paramLengths = new ParamLength[] { 
                new ParamLength(4, 10, 0), 
                new ParamLength(4, 10, 0) 
            };
            int[] paramDirections = new int[] { ParameterDirection.INPUT, ParameterDirection.INPUT };
            Object[] values = new Object[] { 
                stateId <= 0 ? null : stateId, 
                centerId <= 0 ? null : centerId 
            };
            
            objDatabase = new Database();
            ResultSet[] resultSetWrapper = new ResultSet[1];
            
            procReturnValue = objDatabase.execProcedure(
                connectionString, procedureName, params, paramTypes, paramLengths, 
                paramDirections, values, allParameters, resultSetWrapper, true);
            
            objResultSet = resultSetWrapper[0];
        } catch (Exception ex) {
            throw ex;
        } finally {
            objDatabase = null;
        }
        
        return objResultSet;
    }
    
    public ResultSet examCentersForDownload(String connectionString) throws Exception {
        Database objDatabase = null;
        ResultSet objResultSet = null;
        int procReturnValue = 0;
        List<Object> allParameters = new ArrayList<>();
        
        try {
            String procedureName = "sp_get_examcenters4";
            String[] params = new String[] {};
            int[] paramTypes = new int[] {};
            ParamLength[] paramLengths = new ParamLength[] {};
            int[] paramDirections = new int[] {};
            Object[] values = new Object[] {};
            
            objDatabase = new Database();
            ResultSet[] resultSetWrapper = new ResultSet[1];
            
            procReturnValue = objDatabase.execProcedure(
                connectionString, procedureName, params, paramTypes, paramLengths, 
                paramDirections, values, allParameters, resultSetWrapper, true);
            
            objResultSet = resultSetWrapper[0];
        } catch (Exception ex) {
            throw ex;
        } finally {
            objDatabase = null;
        }
        
        return objResultSet;
    }
    
    public ResultSet similarExamCenters(String connectionString, int centerId) throws Exception {
        Database objDatabase = null;
        ResultSet objResultSet = null;
        int procReturnValue = 0;
        List<Object> allParameters = new ArrayList<>();
        
        try {
            String procedureName = "sp_get_examcenters3";
            String[] params = new String[] { "@centerid" };
            int[] paramTypes = new int[] { java.sql.Types.INTEGER };
            ParamLength[] paramLengths = new ParamLength[] { new ParamLength(4, 10, 0) };
            int[] paramDirections = new int[] { ParameterDirection.INPUT };
            Object[] values = new Object[] { centerId };
            
            objDatabase = new Database();
            ResultSet[] resultSetWrapper = new ResultSet[1];
            
            procReturnValue = objDatabase.execProcedure(
                connectionString, procedureName, params, paramTypes, paramLengths, 
                paramDirections, values, allParameters, resultSetWrapper, true);
            
            objResultSet = resultSetWrapper[0];
        } catch (Exception ex) {
            throw ex;
        } finally {
            objDatabase = null;
        }
        
        return objResultSet;
    }
    
    public String saveCenterDetails(String connectionString, int centerId, String centerName, 
                                  String centerCode, int cssCode, int examBody, 
                                  String addressLine1, String addressLine2, String addressLine3, 
                                  int districtId, int pincode, boolean isActive, 
                                  String centerType, int currentUser) throws Exception {
        
        Database objDatabase = null;
        int procReturnValue = 0;
        List<Object> allParameters = new ArrayList<>();
        String message = "";

        try {
            String procedureName = "STP_ADM_SaveExamCenterDetails_New";
            String[] params = new String[] { 
                "@sntExamCenterID", "@varExamCenterName", "@varExamCenterCode", 
                "@tntExamBodyID", "@varHouseNo", "@varStreet", "@varTown", 
                "@sntDistrictID", "@intPinCode", "@btIsActive", "@CurrentUser", 
                "@css_id", "@center_type", "@message" 
            };
            
            int[] paramTypes = new int[] { 
                java.sql.Types.SMALLINT, java.sql.Types.VARCHAR, java.sql.Types.VARCHAR, 
                java.sql.Types.TINYINT, java.sql.Types.VARCHAR, java.sql.Types.VARCHAR, 
                java.sql.Types.VARCHAR, java.sql.Types.SMALLINT, java.sql.Types.INTEGER, 
                java.sql.Types.INTEGER, java.sql.Types.INTEGER, java.sql.Types.INTEGER, 
                java.sql.Types.VARCHAR, java.sql.Types.VARCHAR 
            };
            
            ParamLength[] paramLengths = new ParamLength[] { 
                new ParamLength(2, 5, 0), new ParamLength(256, 0, 0), new ParamLength(5, 0, 0), 
                new ParamLength(1, 3, 0), new ParamLength(256, 0, 0), new ParamLength(256, 0, 0), 
                new ParamLength(256, 0, 0), new ParamLength(2, 5, 0), new ParamLength(4, 10, 0), 
                new ParamLength(4, 10, 0), new ParamLength(4, 10, 0), new ParamLength(4, 10, 0), 
                new ParamLength(20, 0, 0), new ParamLength(255, 0, 0) 
            };
            
            int[] paramDirections = new int[] { 
                ParameterDirection.INPUT, ParameterDirection.INPUT, ParameterDirection.INPUT, 
                ParameterDirection.INPUT, ParameterDirection.INPUT, ParameterDirection.INPUT, 
                ParameterDirection.INPUT, ParameterDirection.INPUT, ParameterDirection.INPUT, 
                ParameterDirection.INPUT, ParameterDirection.INPUT, ParameterDirection.INPUT, 
                ParameterDirection.INPUT, ParameterDirection.OUTPUT 
            };
            
            Object[] values = new Object[] { 
                centerId, centerName, centerCode, examBody, addressLine1, addressLine2, 
                addressLine3, districtId, pincode, isActive ? 1 : 0, currentUser, 
                cssCode, centerType, null 
            };
            
            objDatabase = new Database();
            ResultSet[] resultSetWrapper = new ResultSet[1];
            
            procReturnValue = objDatabase.execProcedure(
                connectionString, procedureName, params, paramTypes, paramLengths, 
                paramDirections, values, allParameters, resultSetWrapper, false);
            
            message = (String) allParameters.get(13);
            
        } catch (Exception ex) {
            throw ex;
        } finally {
            objDatabase = null;
        }
        
        return message;
    }
}

File: source_project/iiiexams.org/IIIDL/ExamCenters.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Data;
using Databases.SQLServer;

namespace IIIDL
{
    public class ExamCenters
    {
        public DataSet FindNearestExamCenter(System.String ConnectionString, System.Int32 Pincode, Int32 ExamBodyId =5 /*for nseit - III online exams*/)
        {
            Databases.SQLServer.Database objDatabase = null;
            DataSet objDataSet = null; //Comment this line if the procedure does not return data set.;
            Int32 ProcReturnValue = 0;
            Object[] AllParameters = null;
            try
            {
                System.String ProcedureName = "STP_CMN_FindNearestExamCenter2";
                String[] Params = new String[] { "@intPINCode", "@tntExamBodyID" };
                SqlDbType[] ParamTypes = new SqlDbType[] { SqlDbType.Int, SqlDbType.TinyInt };
                ParamLength[] ParamLengths = new ParamLength[] { new ParamLength(4, 10, 0), new ParamLength(1, 3, 0) };
                ParameterDirection[] ParamDirections = new ParameterDirection[] { ParameterDirection.Input, ParameterDirection.Input };
                Object[] Values = new Object[] { Pincode, ExamBodyId };
                objDatabase = new Database();
                //Comment the below line if the procedure does not return data set.
                ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters, out objDataSet, true);
                //Comment the below line if the procedure return data set.
                //ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters);
            }
            catch (Exception ex)
            {
                throw (ex);
            }
            finally
            {
                objDatabase = null;
            }
            //Comment the below line if the procedure does not return data set.
            return objDataSet;
        }

        public DataSet GetExamCenters(System.String ConnectionString, System.Int32 Pincode, Int32 ExamBodyId = 5 /*for nseit - III online exams*/)
        {
            Databases.SQLServer.Database objDatabase = null;
            DataSet objDataSet = null; //Comment this line if the procedure does not return data set.;
            Int32 ProcReturnValue = 0;
            Object[] AllParameters = null;
            try
            {
                System.String ProcedureName = "STP_CMN_FindNearestExamCenter3";
                String[] Params = new String[] { "@intPINCode", "@tntExamBodyID" };
                SqlDbType[] ParamTypes = new SqlDbType[] { SqlDbType.Int, SqlDbType.TinyInt };
                ParamLength[] ParamLengths = new ParamLength[] { new ParamLength(4, 10, 0), new ParamLength(1, 3, 0) };
                ParameterDirection[] ParamDirections = new ParameterDirection[] { ParameterDirection.Input, ParameterDirection.Input };
                Object[] Values = new Object[] { Pincode, ExamBodyId };
                objDatabase = new Database();
                //Comment the below line if the procedure does not return data set.
                ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters, out objDataSet, true);
                //Comment the below line if the procedure return data set.
                //ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters);
            }
            catch (Exception ex)
            {
                throw (ex);
            }
            finally
            {
                objDatabase = null;
            }
            //Comment the below line if the procedure does not return data set.
            return objDataSet;
        }

        public DataSet ExamCentersForState(System.String ConnectionString, System.Int32 StateId, Boolean IsTbxCenter)
        {
            Databases.SQLServer.Database objDatabase = null;
            DataSet objDataSet = null; //Comment this line if the procedure does not return data set.;
            Int32 ProcReturnValue = 0;
            Object[] AllParameters = null;
	        try
	        {
		        System.String ProcedureName = "sp_get_examcenters";
                String[] Params = new String[] { "@stateid", "@IsTBXCenter" };
                SqlDbType[] ParamTypes = new SqlDbType[] { SqlDbType.Int, SqlDbType.VarChar };
                ParamLength[] ParamLengths = new ParamLength[] { new ParamLength(4, 10, 0) , new ParamLength(1,0,0) };
                ParameterDirection[] ParamDirections = new ParameterDirection[] { ParameterDirection.Input , ParameterDirection.Input};
                Object[] Values = new Object[] { StateId , IsTbxCenter ? "Y" : "N" };
                objDatabase = new Database();
                //Comment the below line if the procedure does not return data set.
                ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters, out objDataSet, true);
                //Comment the below line if the procedure return data set.
		        //ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters);
	        }
	        catch (Exception ex)
	        {
		        throw(ex);
	        }
	        finally
	        {
		        objDatabase = null;
	        }
            //Comment the below line if the procedure does not return data set.
	        return objDataSet;
        }

        public DataSet ExamCentersForState(System.String ConnectionString, System.Int32 StateId, Int32 CenterId)
        {
            Databases.SQLServer.Database objDatabase = null;
            DataSet objDataSet = null; //Comment this line if the procedure does not return data set.;
            Int32 ProcReturnValue = 0;
            Object[] AllParameters = null;
            try
            {
                System.String ProcedureName = "sp_get_examcenters2";
                String[] Params = new String[] { "@stateid", "@centerid" };
                SqlDbType[] ParamTypes = new SqlDbType[] { SqlDbType.Int, SqlDbType.Int };
                ParamLength[] ParamLengths = new ParamLength[] { new ParamLength(4, 10, 0), new ParamLength(4, 10, 0) };
                ParameterDirection[] ParamDirections = new ParameterDirection[] { ParameterDirection.Input, ParameterDirection.Input };
                Object[] Values = new Object[] { StateId <= 0 ? (Object)DBNull.Value : StateId, CenterId <= 0 ? (Object)DBNull.Value : CenterId };
                objDatabase = new Database();
                //Comment the below line if the procedure does not return data set.
                ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters, out objDataSet, true);
                //Comment the below line if the procedure return data set.
                //ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters);
            }
            catch (Exception ex)
            {
                throw (ex);
            }
            finally
            {
                objDatabase = null;
            }
            //Comment the below line if the procedure does not return data set.
            return objDataSet;
        }

        public DataSet ExamCentersForStateEx(System.String ConnectionString, System.Int32 StateId, Int32 CenterId)
        {
            Databases.SQLServer.Database objDatabase = null;
            DataSet objDataSet = null; //Comment this line if the procedure does not return data set.;
            Int32 ProcReturnValue = 0;
            Object[] AllParameters = null;
            try
            {
                System.String ProcedureName = "sp_get_examcenters5";
                String[] Params = new String[] { "@stateid", "@centerid" };
                SqlDbType[] ParamTypes = new SqlDbType[] { SqlDbType.Int, SqlDbType.Int };
                ParamLength[] ParamLengths = new ParamLength[] { new ParamLength(4, 10, 0), new ParamLength(4, 10, 0) };
                ParameterDirection[] ParamDirections = new ParameterDirection[] { ParameterDirection.Input, ParameterDirection.Input };
                Object[] Values = new Object[] { StateId <= 0 ? (Object)DBNull.Value : StateId, CenterId <= 0 ? (Object)DBNull.Value : CenterId };
                objDatabase = new Database();
                //Comment the below line if the procedure does not return data set.
                ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters, out objDataSet, true);
                //Comment the below line if the procedure return data set.
                //ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters);
            }
            catch (Exception ex)
            {
                throw (ex);
            }
            finally
            {
                objDatabase = null;
            }
            //Comment the below line if the procedure does not return data set.
            return objDataSet;
        }


        public DataSet ExamCentersForDownload(System.String ConnectionString)
        {
            Databases.SQLServer.Database objDatabase = null;
            DataSet objDataSet = null; //Comment this line if the procedure does not return data set.;
            Int32 ProcReturnValue = 0;
            Object[] AllParameters = null;
            try
            {
                System.String ProcedureName = "sp_get_examcenters4";
                String[] Params = new String[] {};
                SqlDbType[] ParamTypes = new SqlDbType[] {};
                ParamLength[] ParamLengths = new ParamLength[] {};
                ParameterDirection[] ParamDirections = new ParameterDirection[] {};
                Object[] Values = new Object[] {};
                objDatabase = new Database();
                //Comment the below line if the procedure does not return data set.
                ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters, out objDataSet, true);
                //Comment the below line if the procedure return data set.
                //ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters);
            }
            catch (Exception ex)
            {
                throw (ex);
            }
            finally
            {
                objDatabase = null;
            }
            //Comment the below line if the procedure does not return data set.
            return objDataSet;
        }

        public DataSet SimilarExamCenters(System.String ConnectionString, Int32 CenterId)
        {
            Databases.SQLServer.Database objDatabase = null;
            DataSet objDataSet = null; //Comment this line if the procedure does not return data set.;
            Int32 ProcReturnValue = 0;
            Object[] AllParameters = null;
            try
            {
                System.String ProcedureName = "sp_get_examcenters3";
                String[] Params = new String[] { "@centerid" };
                SqlDbType[] ParamTypes = new SqlDbType[] { SqlDbType.Int };
                ParamLength[] ParamLengths = new ParamLength[] { new ParamLength(4, 10, 0) };
                ParameterDirection[] ParamDirections = new ParameterDirection[] { ParameterDirection.Input };
                Object[] Values = new Object[] { CenterId };
                objDatabase = new Database();
                //Comment the below line if the procedure does not return data set.
                ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters, out objDataSet, true);
                //Comment the below line if the procedure return data set.
                //ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters);
            }
            catch (Exception ex)
            {
                throw (ex);
            }
            finally
            {
                objDatabase = null;
            }
            //Comment the below line if the procedure does not return data set.
            return objDataSet;
        }

        public String SaveCenterDetails(System.String ConnectionString, System.Int32 CenterId, System.String CenterName, System.String CenterCode, Int32 CSSCode, System.Int32 ExamBody, System.String AddressLine1, System.String AddressLine2, System.String AddressLine3, System.Int32 DistrictId, System.Int32 Pincode, System.Boolean IsActive, String CenterType, System.Int32 CurrentUser)
        {
            Databases.SQLServer.Database objDatabase = null;
            Int32 ProcReturnValue = 0;
            Object[] AllParameters = null;
            String Message = String.Empty;

            try
	        {
                System.String ProcedureName = "STP_ADM_SaveExamCenterDetails_New";
                String[] Params = new String[] { "@sntExamCenterID", "@varExamCenterName", "@varExamCenterCode", "@tntExamBodyID", "@varHouseNo", "@varStreet", "@varTown", "@sntDistrictID", "@intPinCode", "@btIsActive", "@CurrentUser", "@css_id", "@center_type", "@message" };
                SqlDbType[] ParamTypes = new SqlDbType[] { SqlDbType.SmallInt, SqlDbType.VarChar, SqlDbType.VarChar, SqlDbType.TinyInt, SqlDbType.VarChar, SqlDbType.VarChar, SqlDbType.VarChar, SqlDbType.SmallInt, SqlDbType.Int, SqlDbType.Int, SqlDbType.Int, SqlDbType.Int, SqlDbType.VarChar,  SqlDbType.VarChar };
                ParamLength[] ParamLengths = new ParamLength[] { new ParamLength(2, 5, 0), new ParamLength(256, 0, 0), new ParamLength(5, 0, 0), new ParamLength(1, 3, 0), new ParamLength(256, 0, 0), new ParamLength(256, 0, 0), new ParamLength(256, 0, 0), new ParamLength(2, 5, 0), new ParamLength(4, 10, 0), new ParamLength(4, 10, 0), new ParamLength(4, 10, 0), new ParamLength(4, 10, 0), new ParamLength(20, 0, 0),  new ParamLength(255, 0, 0) };
                ParameterDirection[] ParamDirections = new ParameterDirection[] { ParameterDirection.Input, ParameterDirection.Input, ParameterDirection.Input, ParameterDirection.Input, ParameterDirection.Input, ParameterDirection.Input, ParameterDirection.Input, ParameterDirection.Input, ParameterDirection.Input, ParameterDirection.Input, ParameterDirection.Input, ParameterDirection.Input, ParameterDirection.Input, ParameterDirection.Output };
                Object[] Values = new Object[] { CenterId, CenterName, CenterCode, ExamBody, AddressLine1 , AddressLine2, AddressLine3, DistrictId, Pincode, IsActive, CurrentUser, CSSCode, CenterType, DBNull.Value };
                objDatabase = new Database();
                //Comment the below line if the procedure does not return data set.
                //ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters, out objDataSet);
                //Comment the below line if the procedure return data set.
		        ProcReturnValue = objDatabase.ExecProcedure(ConnectionString, ProcedureName, Params, ParamTypes, ParamLengths, ParamDirections, Values, out AllParameters, true);
                Message = Convert.ToString(AllParameters[13]);

            }
	        catch (Exception ex)
	        {
		        throw(ex);
	        }
	        finally
	        {
		        objDatabase = null;
	        }
            //Comment the below line if the procedure does not return data set.
	        return Message;
        }
    }
}
